org.eclipse.xtext/src/org/eclipse/xtext/resource/IMimeTypeProvider.java

```diff
@@ -7,23 +7,23 @@
  */
 package org.eclipse.xtext.resource;
 
-import com.google.inject.ImplementedBy;
-import java.util.Collections;
+import static java.util.Collections.*;
+
 import java.util.List;
-import org.eclipse.xtext.xbase.lib.CollectionLiterals;
+
+import com.google.inject.ImplementedBy;
 
 /**
  * @author Sven Efftinge - Initial contribution and API
  */
 @ImplementedBy(IMimeTypeProvider.DefaultImpl.class)
-@SuppressWarnings("all")
 public interface IMimeTypeProvider {
-  public static class DefaultImpl implements IMimeTypeProvider {
-    @Override
-    public List<String> getMimeTypes() {
-      return Collections.<String>unmodifiableList(CollectionLiterals.<String>newArrayList());
-    }
-  }
-  
-  public abstract List<String> getMimeTypes();
+	List<String> getMimeTypes();
+
+	class DefaultImpl implements IMimeTypeProvider {
```

Comment:
no more need for public?

Reply:
public is implicit in interfaces, and static for classes too